### PR TITLE
Add test executer for testing trainer methods and federated strategies  Methods

### PR DESCRIPTION
**Pull request by ramesht007**
# Description 
Solves Issue #42

### CODE REVIEW

File: configs/dlrm_fl.yml [hunk 1]
```diff
@@ -29,7 +29,7 @@ model :
 
   
   preproc :
-    datafile : "/home/vkkhare/datablaze/criteo_dataset/train.txt"
+    datafile : "/home/ramesht.shuklae/fed/data/train.txt"
```
[the remark this answers is not in the document]
Comment (ramesht007) [reply configs/dlrm_fl.yml:32]: any suggestions for this?

File: fl_strategies/fed_avg.py [hunk 2]
```diff
@@ -18,21 +18,20 @@ def __init__(self,
...
         return arg1 + arg2
 
-    def aggregate(self, neighbour_ids):
-        model_list = [
-            (self.in_neighbours[id].sample_num, self.in_neighbours[id].model)
-            for id in neighbour_ids
-        ]
-        (num0, averaged_params) = model_list[0]
+    def aggregate(self):
+        model_list = [None] * len(self.in_neighbours)
```
[the remark this answers is not in the document]
Comment (ramesht007) [reply fl_strategies/fed_avg.py:22]: cool

File: fedrec/python_executors/trainer.py [hunk 3]
```diff
@@ -94,7 +94,9 @@ def serialize(self):
             round_idx=self.round_idx,
             state_dict=state,
             model_preproc=self.model_preproc,
-            storage=self.persistent_storage
+            storage=self.persistent_storage,
```
[the remark this answers is not in the document]
Comment (ramesht007) [reply fedrec/python_executors/trainer.py:97]: hmm ok

File: configs/dlrm_fl.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (ramesht007) [reply configs/dlrm_fl.yml:32]: cool

**Comment by vkkhare** [comment]
rebase needed

**Pull request closed without merging**